lib/ably/rest/channel.rb

```diff
@@ -65,17 +64,25 @@ def initialize(client, name, channel_options = {})
       #   ]
       #   channel.publish messages
       #
-      def publish(name, data = nil, attributes = {})
-        messages = if name.kind_of?(Enumerable)
-          name
+      #   # Publish a single Ably::Models::Message object, with a query params
+      #   # specifying quickAck: true
+      #   message = Ably::Models::Message(name: 'click', { x: 1, y: 2 })
+      #   channel.publish message, {quickAck: 'true'}
+      #
+      def publish(first, second = nil, third = {})
```

Comment:
`name, data, attributes` made a lot more sense than first, second, third, to be honest. Can we keep those?

Reply:
My reasoning was they make more sense for one out of the three overloaded methods, but are completely wrong for the other two, wheras deliberately-meaningless names act as a reminder that the parameters can be multiple things. WDYT?

Reply:
@SimonWoolf I understand that `first` can be multiple things, but what about `second`/`third`. It's still data and attributes, is it not?

Reply:
no - second/third are only data/attributes respectively if using the `publish(name, data)` form. If using the `publish(Message | Message[], params)` form then `second` is query params, not data payload, and `third` is ignored

Reply:
Cool then, thanks!